### PR TITLE
Update github.com/Microsoft/go-rustaudit to have correct capitalization

### LINKED ISSUE #2026: Cant use syft with go mod vendor because invalid case of microsoft module
**What happened**:
in code you used github.com/microsoft/go-rustaudit instead of github.com/**Microsoft**/go-rustaudit

**What you expected to happen**:
to use github.com/Microsoft/go-rustaudit

**Steps to reproduce the issue**:
install syft as dependency of another go project
run
go mod tidy
go mod vendor
go build -mod vendor

**Anything else we need to know?**:

**Environment**:
- Output of `syft version`:
- OS (e.g: `cat /etc/os-release` or similar):

### PR DESCRIPTION
**Pull request by barnuri**
fixes #2026

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/anchore/syft
 
 go 1.19
 
+replace github.com/microsoft/go-rustaudit => github.com/Microsoft/go-rustaudit v0.0.0-20220730194248-4b17361d90a5
```
Comment (kzantow) [inline go.mod:5]: We definitely don't want to have a replace directive in the `go.mod` -- what is this needed for?

Comment (kzantow) [reply go.mod:5]: This PR has updated the capitalization to be correct, which includes the `go.mod` entries. Why is the `replace` directive needed?

Comment (wagoodman) [reply go.mod]: agreed, the replace directive is not required if we change the import in the source files that are using this module -- just pushed an update that does this

**Comment by wagoodman** [comment]
I'm not certain I see the reason for this change, see https://github.com/anchore/syft/issues/2026#issuecomment-1690518945

**Comment by spiffcs** [comment]
Closing this PR now as we cannot reproduce the issue on our end and have validated that the casing for this is correct at the moment - https://github.com/anchore/syft/issues/2026#issuecomment-1690518945

**Pull request closed without merging**